presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -830,4 +855,27 @@ else if (equality.getRight().equals(column) && columns.contains(equality.getLeft
 
         return Optional.empty();
     }
+
+    private static Optional<Symbol> rewriteExpression(Map<Symbol, Expression> assignments, Expression expression)
+    {
+        checkArgument(expression instanceof CoalesceExpression, "The rewrite can only handle CoalesceExpression");
+        Set<Expression> expressionOperands = ImmutableSet.copyOf(((CoalesceExpression) expression).getOperands());
+        checkArgument(expressionOperands.stream().allMatch(SymbolReference.class::isInstance), "Expect operands of CoalesceExpression to be SymbolReference");
```

Comment:
Let's swap this line and the previous one to make sure that we create set on `SymbolReference`. Also let's declare set as `Set<SymbolReference>`, e.g.: `coleaseExpression.getOperands().stream().map(SymbolReference::cast).collect(toImmutableSet())`